cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -195,15 +258,41 @@ impl BuildEnvNix {
     /// this function is currently assumes that the package is from the nixpkgs base-catalog.
     /// Currently the type is distinguished by the [LockedPackageCatalog::locked_url].
     /// If this does not indicate a nixpkgs package, the function will currently panic!
-    fn realise_nixpkgs(&self, locked: &LockedPackageCatalog) -> Result<(), BuildEnvError> {
-        // check if all store paths are valid, or can be substituted
-        // if so, return without eval
+    fn realise_nixpkgs(
+        &self,
+        client: &impl ClientTrait,
+        locked: &LockedPackageCatalog,
+    ) -> Result<(), BuildEnvError> {
+        // Check if all store paths are valid, or can be substituted.
         let all_valid = self.check_store_path_with_substituters(locked.outputs.values())?;
 
+        // If so, return without eval.
         if all_valid {
             return Ok(());
         }
 
+        let installable = {
+            let mut locked_url = locked.locked_url.to_string();
+
+            if let Some(revision_suffix) = locked_url.strip_prefix(NIXPKGS_CATALOG_URL_PREFIX) {
+                locked_url = format!("{FLOX_NIXPKGS_PROXY_FLAKE_REF_BASE}/{revision_suffix}");
+            } else {
+                debug!(?locked.attr_path, "Trying to realize custom package");
+                let all_found = Self::try_realise_custom_pkg(client, locked)?;
+                // We asked for all the outputs for the package, got store info for
+                // each, and were able to download them all.  If so, then we're done here.
+                if all_found {
+                    return Ok(());
+                };
+                // TODO - Move on to trying to build locally.
+            }
```

Comment:
I agree.  We could move the locked_url prefix check up to `realise_lockfile` but since that's all currently just doing a type `match` that feels hacky.  Rather than doing that, I think I'll update the todo as suggested and file a follow up ticket to start discussing _how_ to differentiate between them and how that migration should happen.

Reply:
> (is custom package our designated term for these now 🤔)

We should clarify, I think I've been using _user_ and _custom_ packages interchangeably.  And I'm not sure either of those is exactly what we want.  I'll ask on the publish channel to clarify for us all.

Reply:
Yes, I agree.  The differentiation currently is by type in a `match` and rather than moving the heuristic check on the locked_url higher up into `realise_lockfile`, I think it would be better to create a follow up ticket to discuss this and refactor accordingly.  I'll update the `panic!` as suggested.